File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-3.py
class Solution:
    def search(self, nums, target):
        left, right = 0, len(nums) - 1
        res = -1
        while left <= right:
            mid = left + (right - left) // 2
            if nums[mid] == target:
                res = mid
                break
            
            if nums[left] <= nums[mid]:
                if target < nums[left] or target > nums[mid]:
                    left = mid + 1
                else:
                    right = mid - 1
            else:
                if target > nums[right] or target < nums[mid]:
                    right = mid - 1
                else:
                    left = mid + 1
        return res

        

File: Data Structures & Algorithms/binary-search/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let lower = 0
        let upper = nums.length -1
        while (lower <= upper) {
            let mid = lower + Math.floor((lower-lower) / 2)
            if (nums[mid] < target) {
                lower = mid + 1
            } else if (nums[mid] > target) {
                upper = mid - 1
            } else {
                return mid
            }
        }
        return -1
    }
}
